core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/recoverer.go

```diff
@@ -112,34 +114,40 @@ func (r *logRecoverer) Start(pctx context.Context) error {
 	r.lock.Unlock()
 
 	blockTimeResolver := newBlockTimeResolver(r.poller)
-	blockTime, err := blockTimeResolver.BlockTime(ctx, defaultSampleSize)
-	if err != nil {
-		// TODO: TBD exit or just log a warning
-		// return fmt.Errorf("failed to compute block time: %w", err)
-		r.lggr.Warnw("failed to compute block time", "err", err)
-	}
-	if blockTime > 0 {
-		r.blockTime.Store(int64(blockTime))
+	updateBlockTime := func(ctx context.Context) {
```

Comment:
Could this function live on the logRecoverer, i.e. could it have a `(r *logRecoverer)` receiver, and take the blockTimeResolver as a param?